src/librustc_typeck/astconv.rs

```diff
@@ -228,14 +227,28 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
         });
 
         if explicit && impl_trait {
+            let spans =
+                seg.generic_args().args
```

Comment:
This may come from my lack of understanding of what exactly there error is here, but would it be possible that we'd want to mark some of these spans and not others?

That is, above, we check if any of the generic args match some condition, but we don't perform a similar check here.  I'm not sure I could construct an example, but could it ever be the case that only a subset of these generic parameters was violating the condition we're checking for above?

Reply:
When I looked at tackling this, I was going to keep track of the indices which matched the pattern above, and then use those same indices to grab the spans in the same way you did here.

Reply:
This error is emitted when we have an `impl Trait` argument, which generates an implicit generic type parameter, and then we try to provide explicit generic arguments — that is, we can't provide any explicit arguments to something that takes an `impl Trait` argument. In that case, it's correct to highlight every single argument, as all of them are offending.

Reply:
Ahh, ok, that makes sense!  Thanks for explaining!